src/tiledb/cloud/vector_search/file_ingestion.py

```diff
@@ -0,0 +1,326 @@
+from typing import Dict, List, Optional, Sequence
+
+from tiledb.cloud import dag
+from tiledb.cloud.utilities import as_batch
+
+
+def ingest_files_dag(
+    file_dir_uri: str,
+    index_uri: str,
+    file_name: Optional[str] = None,
+    acn: Optional[str] = None,
+    config=None,
+    namespace: Optional[str] = None,
+    verbose: bool = False,
+    trace_id: Optional[str] = None,
+    # Index creation params
+    create_index: bool = True,
+    index_type: str = "IVF_FLAT",
+    index_creation_kwargs: Dict = {},
+    # DirectoryTextReader params
+    glob: str = "**/[!.]*",
+    exclude: Sequence[str] = (),
+    suffixes: Optional[Sequence[str]] = None,
+    text_splitter: str = "RecursiveCharacterTextSplitter",
+    text_splitter_kwargs: Optional[Dict] = {
+        "chunk_size": 500,
+        "chunk_overlap": 50,
+    },
+    # SentenceTransformersEmbedding params
+    model_name_or_path: str = "BAAI/bge-small-en-v1.5",
```

Comment:
We had discussed using openai as the default model and having openai api token passed in. What has prompted this change?

Reply:
OpenAi embeddings are nor supported yet in the object api ingestion. If we want OpenAI embeddings we will need to add an implementation and also figure out how to pass the credentials to the taskgraph.

Reply:
The original plan that was discussed was to pass the OpenAI API token as a parameter to start with. Do you feel that is longer viable?

Reply:
In this case, users need to pass it every time the index a file in plain text. We don't have a way to store and retrieve this. 

Apart from the key we need to implement the embedding function that would execute the API calls

Reply:
This sounds like what we had discussed in planning two weeks ago. I suggest we move forward with supporting openai.

Reply:
Added support for OpenAI embeddings